### PR TITLE
[Fix #331] Upgrade to scalaz-specs2 0.4.0

### LINKED ISSUE #331: Upgrade to Specs2-3.x
It's easier to stay current than to stagnate.  That's the upgrade we missed last time.

### PR DESCRIPTION
**Pull request by jedesah**

### CODE REVIEW

**Comment by jedesah** [comment]
Fixes issue #331

File: core/src/test/scala/org/http4s/UrlFormSpec.scala [hunk 1]
```diff
@@ -70,7 +66,7 @@ class UrlFormSpec extends Http4sSpec with ScalaCheck {
       UrlForm(Map("key" -> Seq("a", "b", "c"))).getFirstOrElse("notFound", "d") must_== "d"
     }
 
-    "construct consistently from kv-pairs or and Map[String, Seq[String]]" in check {
+    "construct consistently from kv-pairs or and Map[String, Seq[String]]" in prop {
```
Comment (bryce-anderson) [inline core/src/test/scala/org/http4s/UrlFormSpec.scala]: I don't know why, but this isn't compiling due to a diverging implicit expansion. Any thoughts?

**Comment by bryce-anderson** [comment]
Thanks to tackling this. Its hard to dig up the motivation to refactor tests. :smile: 

This looks good to me once we can get it to compile and pass Travis. That diverging implicit expansion is a bit of a mystery to me. I noticed quite a few instances of explicitly naming the `apply` method: is that something from your IDE or just a personal preference? I don't care either way, I'm just curious.

**Comment by jedesah** [comment]
The use of the `apply` method is something I had to add in order for the tests to compile using specs2 3.0. I am not 100% sure, but I believe it is because a `Service` is a `Kleisli[Task, A, B]` and the `apply` comes from an implicit conversion to `Function1`. Some implicit from specs2 is interfering with that conversion.

File: core/src/test/scala/org/http4s/UrlFormSpec.scala (hunk 1 above)
Comment (jedesah) [inline core/src/test/scala/org/http4s/UrlFormSpec.scala]: Weird, had to do a `clean test:compile` before getting this on my local machine. It is due to the `NonEmptyList` (i.e. it works fine if I change it to a `List`), but I am having a real hard time figuring out why it suddenly is not working anymore.

**Comment by rossabaker** [comment]
I like it once it compiles.  Thanks!

**Comment by jedesah** [comment]
Interesting, the need to add the annoying `apply` explicitly is a Scala 2.10 only requirement.

**Pull request merged**